### ISSUE TITLE
[Suggestion] Give a setting to disable replacing items on hotbar or at least to not take priority over typing in the search bar

### ISSUE THREAD

**Issue by Keltas** [mezz/JustEnoughItems#2612]
Hello,
I regularly struggle with the issue of randomly replacing items in my hotbar with random junk when in cheat mode. This occurs because pressing shift+number, results in the slot of that number being replaced with the item that you hover over in the JEI. This is problematic due to the fact that "shift+2" is the combination required to type "@" into the search bar. This is especially annoying considering that the aforementioned combination has the priority over the search bar box. 
As such even if the search bar is active and you are in the middle of typing and you press the aforementioned combination, the item in slot 2 gets replaced by whatever my mouse happens to hover over, while deactivating the typing in search bar. I regularly use the "@mod" functionality whereas I've never needed the extra convenience from the faster replacing items in my hot bar.
This functionality has resulted in countless times where I've lost important items as I was not aware that it even existed. For a long time I thought that some unidentifiable mod is simply eating my items and replacing them with random junk.

To summarize I would like to suggest that:
1. For the typing in the search bar, to take priority over the "shift+number" shortcuts as the characters accessed via typing "shift+number" are often necessary for the use of the search bar.
2. For there to be a setting to disable these shortcuts, as from what I have seen they are absent from the controls menu in Minecraft and I could not find them in the config files.

**Comment by mezz** [comment]
Thanks for the report!
Which version of JEI are you using? Hotkeys have been re-written in the latest version and if you're on an older version I'll need to make sure it's fixed there as well.

**Comment by Keltas** [comment]
I was using jei-1.16.5-7.7.1.138 when last encountering it. With Forge version 36.2.20. Although I am pretty sure it occurred in earlier versions as well. I  think it began around 1.15.2 though I am not sure of that.

**Comment by mezz** [comment]
I tested JEI for 1.17 and 1.18 and typing in the search bar takes priority there.
1.16 still has the issue.
I'll add a config for this feature, I think it's a good idea to have it disabled by default.
